Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -154,16 +154,40 @@ GridArea GridMasonryLayout::gridAreaForDefiniteGridAxisItem(const RenderBox& gri
     return masonryGridAreaFromGridAxisSpan(itemSpan);
 }
 
-void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm)
+void GridMasonryLayout::placeItemsWithIndefiniteGridAxisPosition(const GridTrackSizingAlgorithm& algorithm, std::optional<GridMasonryLayout::OverrideMasonryLogicalWidth> overrideLogicalWidthType)
 {
     for (auto* item : m_itemsWithIndefiniteGridAxisPosition) {
         ASSERT(item);
         if (!item)
             continue;
-        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item));
+        insertIntoGridAndLayoutItem(algorithm, *item, gridAreaForIndefiniteGridAxisItem(*item), overrideLogicalWidthType);
     }
 }
 
+LayoutUnit GridMasonryLayout::calculateMasonryIntrinsicLogicalWidth(RenderBox& gridItem, GridMasonryLayout::OverrideMasonryLogicalWidth overrideLogicalWidthType)
+{
+    switch (overrideLogicalWidthType) {
+    case OverrideMasonryLogicalWidth::MinContent:
+        if ((gridItem.style().logicalWidth().isAuto() || gridItem.style().logicalWidth().isPercent())) {
```

Comment:
Maybe also consider pulling this to the top as a way to return as early as possible if this condition is not true?